src/System.Text.Json/tests/JsonElementWriteTests.cs

```diff
@@ -523,6 +526,43 @@ public static void WriteNumberScientificAsProperty(bool indented)
                 "{\"m\\u00eal\\u00e9e\":1e6}");
         }
 
+        [Fact]
+        public static void WriteValueSurrogatesEscapeString()
+        {
+            string unicodeString = "\uD800\uDC00\uD803\uDE6D \uD834\uDD1E\uDBFF\uDFFF";
+            string json = $"[\"{unicodeString}\"]";
+            var buffer = new ArrayBufferWriter<byte>(1024);
+            string expectedStr = GetEscapedExpectedString(unicodeString, StringEscapeHandling.EscapeNonAscii);
+
+            using (JsonDocument doc = JsonDocument.Parse(json, s_options))
+            {
+                JsonElement target = doc.RootElement[0];
+
+                using (var writer = new Utf8JsonWriter(buffer))
+                {
+                    target.WriteTo(writer);
+                    writer.Flush();
+                }
+                AssertContents(expectedStr, buffer);
+            }
+        }
+
+        private static string GetEscapedExpectedString(string value, StringEscapeHandling escaping)
```

Comment:
nit: You could use a `StringWriter` instead, and skip the transcoding to utf-8 and back. But this is outside of the scope of this PR and if we are changing it, we may want to fix it in the existing writer tests as well:
https://github.com/dotnet/corefx/blob/05c0f516cf20cd6ae108ba034984b0cdc740732e/src/System.Text.Json/tests/Utf8JsonWriterTests.cs#L5383-L5904

```C#
private static string GetEscapedExpectedString(string value, StringEscapeHandling escaping)
{
    using (TextWriter stringWriter = new StringWriter())
    using (var json = new JsonTextWriter(stringWriter)
    {
        StringEscapeHandling = escaping
    })
    {
        json.WriteValue(value);
        json.Flush();
        return stringWriter.ToString();
    }
}
```